src/librustc/hir/intravisit.rs

```diff
@@ -1011,22 +1011,22 @@ pub fn walk_expr<'v, V: Visitor<'v>>(visitor: &mut V, expression: &'v Expr) {
             visitor.visit_expr(subexpression);
             visitor.visit_ty(typ)
         }
-        ExprKind::If(ref head_expression, ref if_block, ref optional_else) => {
-            visitor.visit_expr(head_expression);
+        ExprKind::If(ref scrutinee, ref if_block, ref optional_else) => {
```

Comment:
For `If` and `While` this should be `condition` (although I guess this code is going away soon).